### PR TITLE
feat: add support for NodeJS 18 in generated dockerfiles

### PR DESCRIPTION
**Pull request by philasmar**
*Description of changes:*
* Upgrade NodeJS to v18 since v16 reached it's end of life and deployments were showing a deprecation notice.
* Upgraded project dependencies
* Adapted project to allow building it with .NET8 SDK
* Updated the way we were installing NodeJS on our dockerfiles since the script we were using has also been deprecated

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs [hunk 1]
```diff
@@ -512,7 +512,7 @@ public async Task<Vpc> GetDefaultVpc()
                             }
                         }
                     })
-                .Vpcs.FirstOrDefaultAsync(),
+                .Vpcs.FirstAsync(),
```
Comment (normj) [inline src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs:515]: What is the reason for this change? Won't this cause an error message to the user where we just add "Sequence contains no elements" to the end of there are is no default VPC?

**Review by normj — CHANGES_REQUESTED** [review]

File: src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs (hunk 1 above)
Comment (philasmar) [reply src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs]: Building the solution with .NET8 threw a "possible null reference" exception for that block of code. I switched back to `FirstOrDefaultAsync` and instead I am not returning a nullable VPC.

**Review by normj — APPROVED** [review]

File: site/content/docs/getting-started/pre-requisites.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ dotnet --version
 
 #### Node.js
 
-* The deployment tool requires the [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/latest/guide/), and the AWS CDK requires [Node.js](https://nodejs.org/en/download/). AWS CDK requires Node.js, versions 14.x, 16.x, 18.x (or later) - we recommend installing the latest LTS version.
+* The deployment tool requires the [AWS Cloud Development Kit (CDK)](https://docs.aws.amazon.com/cdk/latest/guide/), and the AWS CDK requires [Node.js](https://nodejs.org/en/download/). AWS CDK requires Node.js 18.x (or later) - we recommend installing the latest LTS version.
```
Comment (ashovlin) [inline site/content/docs/getting-started/pre-requisites.md:25]: nit: Does CDK require 18 now? They still say 14 here: https://github.com/aws/aws-cdk#getting-started

Fine if _we_ require it, just don't want to mislead anybody.

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj [hunk 3]
```diff
@@ -47,6 +47,10 @@
     <None Include="..\..\README.md" Pack="true" PackagePath="" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="Nerdbank.GitVersioning" Version="3.6.133" />
```
Comment (ashovlin) [inline src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:51]: Curious, why this addition here now?

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj [hunk 4]
```diff
@@ -21,15 +21,15 @@
...
 
   <ItemGroup>
-    <PackageReference Include="AWSSDK.IdentityManagement" Version="3.7.100.15" />
-    <PackageReference Include="AWSSDK.CloudFormation" Version="3.7.101.9" />
-    <PackageReference Include="AWSSDK.SecurityToken" Version="3.7.100.14" />
-    <PackageReference Include="AWSSDK.SSO" Version="3.7.100.14" />
-    <PackageReference Include="AWSSDK.SSOOIDC" Version="3.7.100.14" />
-    <PackageReference Include="Swashbuckle.AspNetCore.Annotations" Version="6.1.2" />
-    <PackageReference Include="Swashbuckle.AspNetCore.Swagger" Version="6.1.2" />
+    <PackageReference Include="AWSSDK.IdentityManagement" Version="3.7.200.42" />
```
Comment (ashovlin) [inline src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:24]: Could you update https://github.com/aws/aws-dotnet-deploy/blob/main/THIRD_PARTY_LICENSES and the internal tracker as well for all the version bumps?

File: site/content/docs/getting-started/pre-requisites.md (hunk 2 above)
Comment (philasmar) [reply site/content/docs/getting-started/pre-requisites.md]: I updated the wording to not explicitly indicate that CDK is requiring a specific version, rather a specific version is required to use the tool.

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj (hunk 3 above)
Comment (philasmar) [reply src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj]: This was a miss. While using the Nuget Package Manager, it added it to the csproj instead of the `Directory.Build.props` file. Good catch.

File: src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj (hunk 4 above)
Comment (philasmar) [reply src/AWS.Deploy.CLI/AWS.Deploy.CLI.csproj:24]: I went over all the dependencies and updated them. Also removed any dependency not referenced explicitly in the project.

**Review by ashovlin — APPROVED** [review]

**Review by ashovlin — APPROVED** [review]
Looks good after https://github.com/aws/aws-dotnet-deploy/compare/b667a50dc3b63566ef412e6aecc81637ad5d520f..8e839e30eb2a203b08f212be2a11baa49fc48426

**Review by normj — APPROVED** [review]
